[Main] Recognize ddl.to and turb.to Issue 3702

OzzieIsaacs opened the pull request:
### Describe the changes

I applied my suggested change from issue #3702 to make XFelesharing last in the list to compare the regexes
I furthermore updated ddl and turbobit plugin

### Is this related to a problem?

Issue #3702

### Additional references

pep8speaks commented:
Hello @OzzieIsaacs! Thanks for opening this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:

* In the file [`src/pyload/core/managers/plugin_manager.py`](https://github.com/pyload/pyload/blob/3b243b68c33c8e75d79b0a6bd753dde9ca394731/src/pyload/core/managers/plugin_manager.py):

> [Line 253:16](https://github.com/pyload/pyload/blob/3b243b68c33c8e75d79b0a6bd753dde9ca394731/src/pyload/core/managers/plugin_manager.py#L253): [E127](https://duckduckgo.com/?q=pep8%20E127) continuation line over-indented for visual indent

* In the file [`src/pyload/plugins/downloaders/DdlTo.py`](https://github.com/pyload/pyload/blob/3b243b68c33c8e75d79b0a6bd753dde9ca394731/src/pyload/plugins/downloaders/DdlTo.py):

> [Line 9:1](https://github.com/pyload/pyload/blob/3b243b68c33c8e75d79b0a6bd753dde9ca394731/src/pyload/plugins/downloaders/DdlTo.py#L9): [E302](https://duckduckgo.com/?q=pep8%20E302) expected 2 blank lines, found 1
> [Line 18:89](https://github.com/pyload/pyload/blob/3b243b68c33c8e75d79b0a6bd753dde9ca394731/src/pyload/plugins/downloaders/DdlTo.py#L18): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (91 > 88 characters)
> [Line 20:89](https://github.com/pyload/pyload/blob/3b243b68c33c8e75d79b0a6bd753dde9ca394731/src/pyload/plugins/downloaders/DdlTo.py#L20): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (95 > 88 characters)
> [Line 48:89](https://github.com/pyload/pyload/blob/3b243b68c33c8e75d79b0a6bd753dde9ca394731/src/pyload/plugins/downloaders/DdlTo.py#L48): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (104 > 88 characters)

* In the file [`src/pyload/plugins/downloaders/TurbobitNet.py`](https://github.com/pyload/pyload/blob/3b243b68c33c8e75d79b0a6bd753dde9ca394731/src/pyload/plugins/downloaders/TurbobitNet.py):

> [Line 18:89](https://github.com/pyload/pyload/blob/3b243b68c33c8e75d79b0a6bd753dde9ca394731/src/pyload/plugins/downloaders/TurbobitNet.py#L18): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (112 > 88 characters)

We recommend to use [black](https://github.com/python/black) to automatically correct these files.

GammaC0de commented:
Please remove the changes to plugin_manager.py

In my opinion, the filename should not affect the  plugin order.
Also, I intend to make Xfilesharing disabled by default

OzzieIsaacs commented:
Hello @GammaC0de, I'm not ready to give up yet. I agree with you the name is not the best solution to get the plugins ordered. So another proposal:
I've added a member \_\_order\_\_, it can have any number (integer would normally sufficient, I now used float to have the same regex as for the version parser). 
This \_\_order\_\_ variable is added to the dict and used for sorting. The only plugin which needs the order member really set is the XFIleSharing plugin to move it to the end, everything else can stay at default (e.g. 0).

What about this (I already changed the PR accordingly)?

vuolter commented:
As I remember plugins with pattern `^unmatchable$` should already run at the end, if no one else is matched. If it does not happen, there is a bug or some code was accidentally removed during the migration to py3.

OzzieIsaacs commented:
@voulter: I checked it again, it's not working (not for the development branch and also not for stable, as reported in #3702). 
I didn't 100% get what is going on. It's matching with the name of XFilesharing, but the pattern looks like the one from "MultiAccount" (at least in pyload_ng)
